src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -1321,9 +1343,78 @@ private boolean performDecoderControl(DecoderInfo decoderInfo, String controlURL
 		} catch (Exception e) {
 			throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + controlMethod + DecoderConstant.NEXT_LINE + e.getMessage());
 		}
-		return true;
+		return decoderInfoResult;
 	}
 
+	/**
+	 * This method is used to perform decoder control start/ stop/ update
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 * @param decoderInfo set of decoder config info
+	 * @param controlMethod start/stop/update
+	 * @param controlURL API entry point
+
+	 */
+	private void performActiveDecoderControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, DecoderInfo decoderInfo, String controlURL, String controlMethod) {
+		String decoderID = decoderInfo.getId();
+		String decoderControllingGroup = ControllingMetricGroup.DECODER.getName() + decoderID + DecoderConstant.HASH;
+		try {
+			if (this.authenticationCookie.getSessionID() != null) {
+				String request = decoderInfo.jsonRequest();
+				if (logger.isDebugEnabled()) {
+					logger.debug(request);
+				}
+				DecoderData decoderData = doPut(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.DECODERS + DecoderConstant.SLASH + decoderID + controlURL), request, DecoderData.class);
+				if (decoderData == null) {
+					populateActiveControlAfterControlFailed(decoderInfo);
+					addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, decoderControllingGroup + DecoderControllingMetric.STATE.getName(), decoderInfo.getState().isRunning(),
+							DecoderConstant.OFF, DecoderConstant.ON));
+					throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + controlMethod);
+				}else{
+					this.decoderInfoDTOList.set(Integer.parseInt(decoderID), decoderData.getDecoderInfo());
+					this.localDecoderInfoList.set(Integer.parseInt(decoderID), decoderData.getDecoderInfo());
+					populateDecoderControl(stats, advancedControllableProperties, Integer.parseInt(decoderID));
+				}
+			} else {
+				populateActiveControlAfterControlFailed(decoderInfo);
+				addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, decoderControllingGroup + DecoderControllingMetric.STATE.getName(), decoderInfo.getState().isRunning(),
+						DecoderConstant.OFF, DecoderConstant.ON));
+				throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + controlMethod);
+			}
+		} catch (Exception e) {
+			populateActiveControlAfterControlFailed(decoderInfo);
+			addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, decoderControllingGroup + DecoderControllingMetric.STATE.getName(), decoderInfo.getState().isRunning(),
+					DecoderConstant.OFF, DecoderConstant.ON));
+			throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + controlMethod + DecoderConstant.NEXT_LINE + e.getMessage());
```

Comment:
the same issue like above, please do not swallow the exception, add the "e" like second arg

Reply:
i fixed it